---
# Documentation: https://wowchemy.com/docs/managing-content/

title: Resources for Interpreting Variants in Precision Genomic Oncology Applications
subtitle: ''
summary: ''
authors:
- Hsinyi Tsang
- KanakaDurga Addepalli
- Sean R. Davis
tags: []
categories: []
date: '2017-01-01'
lastmod: 2021-05-08T11:20:16-06:00
featured: false
draft: false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: ''
  focal_point: ''
  preview_only: false

# Projects (optional).
#   Associate this post with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects: []
publishDate: '2021-05-08T17:20:16.090857Z'
publication_types:
- '2'
abstract: Precision genomic oncology--applying high throughput sequencing (HTS) at
  the point-of-care to inform clinical decisions--is a developing precision medicine
  paradigm that is seeing increasing adoption. Simultaneously, new developments in
  targeted agents and immunotherapy, when informed by rich ge- nomic characterization,
  offer potential benefit to a growing subset of patients. Multiple previous studies
  have commented on methods for identifying both germline and somatic variants. However,
  interpreting individual variants remains a significant challenge, relying in large
  part on the integration of observed variants with biological knowledge. A number
  of data and software resources have been developed to assist in interpreting observed
  variants, determining their potential clinical actionability, and augmenting them
  with ancillary information that can inform clinical decisions and even generate
  new hypotheses for exploration in the laboratory. Here, we review available variant
  catalogs, variant and functional an- notation software and tools, and databases
  of clinically actionable variants that can be used in an ad hoc approach with research
  samples or incorporated into a data platform for interpreting and formally reporting
  clinical results.
publication: '*Frontiers in Oncology*'
url_pdf: http://journal.frontiersin.org/article/10.3389/fonc.2017.00214
doi: 10.3389/fonc.2017.00214
---
