---
# Documentation: https://wowchemy.com/docs/managing-content/

title: NCI-60 whole exome sequencing and pharmacological CellMiner analyses
subtitle: ''
summary: ''
authors:
- W. C. Reinhold
- S. Varma
- F. Sousa
- M. Sunshine
- O. D. Abaan
- S. R. Davis
- S. W. Reinhold
- K. W. Kohn
- J. Morris
- P. S. Meltzer
- J. H. Doroshow
- Y. Pommier
tags: []
categories: []
date: '2014-01-01'
lastmod: 2021-05-08T11:20:21-06:00
featured: false
draft: false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: ''
  focal_point: ''
  preview_only: false

# Projects (optional).
#   Associate this post with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects: []
publishDate: '2021-05-08T17:20:20.862552Z'
publication_types:
- '2'
abstract: Exome sequencing provides unprecedented insights into cancer biology and
  pharmacological response. Here we assess these two parameters for the NCI-60, which
  is among the richest genomic and pharmacological publicly available cancer cell
  line databases. Homozygous genetic variants that putatively affect protein function
  were identified in 1,199 genes (approximately 6% of all genes). Variants that are
  either enriched or depleted compared to non-cancerous genomes, and thus may be influential
  in cancer progression and differential drug response were identified for 2,546 genes.
  Potential gene knockouts are made available. Assessment of cell line response to
  19,940 compounds, including 110 FDA-approved drugs, reveals â‰ˆ80-fold range in
  resistance versus sensitivity response across cell lines. 103,422 gene variants
  were significantly correlated with at least one compound (at p<0.0002). These include
  genes of known pharmacological importance such as IGF1R, BRAF, RAD52, MTOR, STAT2
  and TSC2 as well as a large number of candidate genes such as NOM1, TLL2, and XDH.
  We introduce two new web-based CellMiner applications that enable exploration of
  variant-to-compound relationships for a broad range of researchers, especially those
  without bioinformatics support. The first tool, \"Genetic variant versus drug visualization\",
  provides a visualization of significant correlations between drug activity-gene
  variant combinations. Examples are given for the known vemurafenib-BRAF, and novel
  ifosfamide-RAD52 pairings. The second, \"Genetic variant summation\" allows an assessment
  of cumulative genetic variations for up to 150 combined genes together; and is designed
  to identify the variant burden for molecular pathways or functional grouping of
  genes. An example of its use is provided for the EGFR-ERBB2 pathway gene variant
  data and the identification of correlated EGFR, ERBB2, MTOR, BRAF, MEK and ERK inhibitors.
  The new tools are implemented as an updated web-based CellMiner version, for which
  the present publication serves as a compendium.
publication: '*PLoS ONE*'
---
