---
# Documentation: https://wowchemy.com/docs/managing-content/

title: Genome-wide mapping of DNase hypersensitive sites using massively parallel
  signature sequencing (MPSS)
subtitle: ''
summary: ''
authors:
- G. E. Crawford
- I. E. Holt
- J. Whittle
- B. D. Webb
- D. Tai
- S. Davis
- E. H. Margulies
- Y. Chen
- J. A. Bernat
- D. Ginsburg
- D. Zhou
- S. Luo
- T. J. Vasicek
- M. J. Daly
- T. G. Wolfsberg
- F. S. Collins
tags: []
categories: []
date: '2006-01-01'
lastmod: 2021-08-21T11:08:48-06:00
featured: false
draft: false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: ''
  focal_point: ''
  preview_only: false

# Projects (optional).
#   Associate this post with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects: []
publishDate: '2021-08-21T20:26:31.136790Z'
publication_types:
- '2'
abstract: A major goal in genomics is to understand how genes are regulated in different
  tissues, stages of development, diseases, and species. Mapping DNase I hypersensitive
  (HS) sites within nuclear chromatin is a powerful and well-established method of
  identifying many different types of regulatory elements, but in the past it has
  been limited to analysis of single loci. We have recently described a protocol to
  generate a genome-wide library of DNase HS sites. Here, we report high-throughput
  analysis, using massively parallel signature sequencing (MPSS), of 230,000 tags
  from a DNase library generated from quiescent human CD4+ T cells. Of the tags that
  uniquely map to the genome, we identified 14,190 clusters of sequences that group
  within close proximity to each other. By using a real-time PCR strategy, we determined
  that the majority of these clusters represent valid DNase HS sites. Approximately
  80% of these DNase HS sites uniquely map within one or more annotated regions of
  the genome believed to contain regulatory elements, including regions 2 kb upstream
  of genes, CpG islands, and highly conserved sequences. Most DNase HS sites identified
  in CD4+ T cells are also HS in CD8+ T cells, B cells, hepatocytes, human umbilical
  vein endothelial cells (HUVECs), and HeLa cells. However, approximately 10% of the
  DNase HS sites are lymphocyte specific, indicating that this procedure can identify
  gene regulatory elements that control cell type specificity. This strategy, which
  can be applied to any cell line or tissue, will enable a better understanding of
  how chromatin structure dictates cell function and fate.
publication: '*Genome Res.*'
---
