---
# Documentation: https://wowchemy.com/docs/managing-content/

title: Whole genome and transcriptome sequencing of a B3 thymoma
subtitle: ''
summary: ''
authors:
- I. Petrini
- A. Rajan
- T. Pham
- D. Voeller
- S. Davis
- J. Gao
- Y. Wang
- G. Giaccone
tags: []
categories: []
date: '2013-01-01'
lastmod: 2021-05-08T11:20:23-06:00
featured: false
draft: false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: ''
  focal_point: ''
  preview_only: false

# Projects (optional).
#   Associate this post with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects: []
publishDate: '2021-05-08T17:20:22.924929Z'
publication_types:
- '2'
abstract: Molecular pathology of thymomas is poorly understood. Genomic aberrations
  are frequently identified in tumors but no extensive sequencing has been reported
  in thymomas. Here we present the first comprehensive view of a B3 thymoma at whole
  genome and transcriptome levels. A 55-year-old Caucasian female underwent complete
  resection of a stage IVA B3 thymoma. RNA and DNA were extracted from a snap frozen
  tumor sample with a fraction of cancer cells over 80%. We performed array comparative
  genomic hybridization using Agilent platform, transcriptome sequencing using HiSeq
  2000 (Illumina) and whole genome sequencing using Complete Genomics Inc platform.
  Whole genome sequencing determined, in tumor and normal, the sequence of both alleles
  in more than 95% of the reference genome (NCBI Build 37). Copy number (CN) aberrations
  were comparable with those previously described for B3 thymomas, with CN gain of
  chromosome 1q, 5, 7 and X and CN loss of 3p, 6, 11q42.2-qter and q13. One translocation
  t(11;X) was identified by whole genome sequencing and confirmed by PCR and Sanger
  sequencing. Ten single nucleotide variations (SNVs) and 2 insertion/deletions (INDELs)
  were identified; these mutations resulted in non-synonymous amino acid changes or
  affected splicing sites. The lack of common cancer-associated mutations in this
  patient suggests that thymomas may evolve through mechanisms distinctive from other
  tumor types, and supports the rationale for additional high-throughput sequencing
  screens to better understand the somatic genetic architecture of thymoma.
publication: '*PLoS ONE*'
---
