---
title: Leveraging Bioconductor for somatic variant analysis of TCGA data
author: Sean Davis
date: '2018-01-26'
slug: leveraging-bioconductor-for-somatic-variant-analysis-of-tcga-dataRmd
categories:
  - Bioconductor
tags:
  - Cancer Genomics
  - GenomicDataCommons
  - maftools
---



<p>The <a href="https://gdc.cancer.gov">NCI Genomic Data Commons (GDC)</a> now contains the authoritative
source of data from The Cancer Genome Atlas (TCGA) as well as several
other projects of import to the cancer research community. One of the
available assays produces somatic variant calls, formally identified
by comparing tumor reads and normal reads to identify variants
relative to the human reference genome that are not present in the
normal genome of the patient. Unfortunately, this process for
discovering these variants is less precise relative to finding
germline variants. Because a gold standard bioinformatics approach for
somatic variant calling has not been identified, the GDC runs four
separate pipelines and has two levels of data for each. This post
demonstrates the use of the Bioconductor <a href="https://bioconductor.org/packages/GenomicDataCommons">GenomicDataCommons package</a>
to identify and then fetch somatic variant results from the NCI GDC
and then provide a high-level assessment of those variants using the
<a href="https://bioconductor.org/packages/maftools">maftools package</a>. The workflow will be:</p>
<ol style="list-style-type: decimal">
<li>Install packages if not already installed</li>
<li>Load libraries</li>
<li>Identify and download somatic variants for a representative TCGA
dataset, in this case cutaneous melanoma.</li>
<li>Use maftools to provide rich summaries of the data.</li>
</ol>
<p>Identifying appropriate resources at the GDC starts with a bit of
browsing on the website or with equivalent browsing using one of the
GenomicDataCommons introspection functions, <code>available_fields()</code> and
<code>available_values()</code>.</p>
<p>To work through these code examples, the maftools and
GenomicDataCommons packages must first be installed.</p>
<pre class="r"><code>source(&quot;https://bioconductor.org/biocLite.R&quot;)
biocLite(c(&quot;GenomicDataCommons&quot;, &quot;maftools&quot;)</code></pre>
<p>Once installed, load the packages, as usual.</p>
<pre class="r"><code>library(GenomicDataCommons)
library(maftools)</code></pre>
<p>The data model (how data are described and linked to each other) is
quite complicated, but documentation is available. In general, though,
linking Bioconductor tools to data involves finding files that meet
some criteria. Each file in the GDC is identified by a “UUID” that
carries no meaning other than it is unique. TCGA barcodes are not
<em>directly</em> used for identifying files, though filtering files by TCGA
barcodes is possible (and not shown here). So, the first step is to
find barcodes associated with a [MAF format file] for TCGA project
“TCGA-SKCM”. Searching based on <code>data_type</code>, <code>data_format</code>, and
<code>analysis.workflow_type</code> will limit results to the file of interest,
namely the <a href="https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&amp;id=somatic_aggregation_workflow&amp;_top=1">MuTect2 workflow</a> variant calls, converted to MAF format.</p>
<pre class="r"><code>uuids = files() %&gt;% 
    GenomicDataCommons::filter(~ cases.project.project_id==&#39;TCGA-SKCM&#39; &amp; 
        data_type==&#39;Masked Somatic Mutation&#39; &amp;
        data_format==&#39;MAF&#39; &amp;
        analysis.workflow_type==&#39;MuTect2 Variant Aggregation and Masking&#39;) %&gt;%
    ids()</code></pre>
<p>Once the uuids have been identified (in this case, only 4b7a5729-b83e-4837-9b61-a6002dce1c0a),
the <code>gdcdata()</code> function downloads the associated files and returns a
filename for each uuid.</p>
<pre class="r"><code>maffile = gdcdata(uuids)</code></pre>
<p>The MAF file is now stored locally and the <a href="https://bioconductor.org/packages/maftools">maftools package</a>
workflow, which starts with a MAF file, can proceed, starting with
reading the melanoma MAF file.</p>
<pre class="r"><code>melanoma_vars = read.maf(maf = maffile, verbose = FALSE)</code></pre>
<p>With the data available as a maftools MAF object, a lot of
functionality is available with little code. While the maftools
package offers quite a few functions, here are a few
highlights. Cancer genomics and bioinformatics researchers will
recognize these plots, most likely.</p>
<pre class="r"><code>plotmafSummary(maf = melanoma_vars, rmOutlier = TRUE,
               addStat = &#39;median&#39;, dashboard = TRUE,
               titvRaw = FALSE)</code></pre>
<p><img src="/post/2018-01-26-leveraging-bioconductor-for-somatic-variant-analysis-of-tcga-data_files/figure-html/mafsummary-1.png" width="672" /></p>
<pre class="r"><code># exclude a couple of &quot;noise&quot; genes--common practice, unfortunately
oncoplot(maf = melanoma_vars, top = 15, fontSize = 12,
         genes = getGeneSummary(melanoma_vars)$Hugo_Symbol[3:20])</code></pre>
<p><img src="/post/2018-01-26-leveraging-bioconductor-for-somatic-variant-analysis-of-tcga-data_files/figure-html/oncoprint-1.png" width="672" /></p>
<pre class="r"><code>braf.lpop = lollipopPlot(maf = melanoma_vars, gene = &#39;BRAF&#39;,
                           AACol = &#39;HGVSp_Short&#39;, showMutationRate = TRUE,
                           domainLabelSize = 3, defaultYaxis = FALSE)</code></pre>
<p>Additional functionality is available for both the GenomicDataCommons
and maftools packages. This post just highlights how one can leverage
two Bioconductor packages to quickly gain insight into public cancer
genomics datasets.</p>
<div id="appendix" class="section level1">
<h1>Appendix</h1>
<p>Examining available values in the GDC is straightforward, allowing for
variations on the workflow above.</p>
<div id="available-data_types" class="section level2">
<h2>Available data_types:</h2>
<pre class="r"><code>available_values(&#39;files&#39;,&#39;data_type&#39;)</code></pre>
<pre><code>##  [1] &quot;Annotated Somatic Mutation&quot;       
##  [2] &quot;Raw Simple Somatic Mutation&quot;      
##  [3] &quot;Aligned Reads&quot;                    
##  [4] &quot;Gene Expression Quantification&quot;   
##  [5] &quot;Slide Image&quot;                      
##  [6] &quot;Biospecimen Supplement&quot;           
##  [7] &quot;Copy Number Segment&quot;              
##  [8] &quot;Masked Copy Number Segment&quot;       
##  [9] &quot;Clinical Supplement&quot;              
## [10] &quot;Methylation Beta Value&quot;           
## [11] &quot;Isoform Expression Quantification&quot;
## [12] &quot;miRNA Expression Quantification&quot;  
## [13] &quot;Raw CGI Variant&quot;                  
## [14] &quot;Aggregated Somatic Mutation&quot;      
## [15] &quot;Masked Somatic Mutation&quot;</code></pre>
</div>
<div id="available-data_formats" class="section level2">
<h2>Available data_formats:</h2>
<pre class="r"><code>available_values(&#39;files&#39;,&#39;data_format&#39;)</code></pre>
<pre><code>##  [1] &quot;VCF&quot;               &quot;TXT&quot;               &quot;BAM&quot;              
##  [4] &quot;SVS&quot;               &quot;BCR XML&quot;           &quot;BCR SSF XML&quot;      
##  [7] &quot;BCR Auxiliary XML&quot; &quot;BCR OMF XML&quot;       &quot;BCR Biotab&quot;       
## [10] &quot;MAF&quot;               &quot;XLSX&quot;</code></pre>
</div>
<div id="available-analysis-workflows" class="section level2">
<h2>Available analysis workflows:</h2>
<pre class="r"><code>available_values(&#39;files&#39;,&#39;analysis.workflow_type&#39;)</code></pre>
<pre><code>##  [1] &quot;DNAcopy&quot;                                      
##  [2] &quot;BCGSC miRNA Profiling&quot;                        
##  [3] &quot;BWA with Mark Duplicates and Cocleaning&quot;      
##  [4] &quot;FM Simple Somatic Mutation&quot;                   
##  [5] &quot;FoundationOne Annotation&quot;                     
##  [6] &quot;Liftover&quot;                                     
##  [7] &quot;STAR 2-Pass&quot;                                  
##  [8] &quot;HTSeq - Counts&quot;                               
##  [9] &quot;HTSeq - FPKM&quot;                                 
## [10] &quot;HTSeq - FPKM-UQ&quot;                              
## [11] &quot;BWA-aln&quot;                                      
## [12] &quot;SomaticSniper&quot;                                
## [13] &quot;SomaticSniper Annotation&quot;                     
## [14] &quot;MuTect2&quot;                                      
## [15] &quot;MuTect2 Annotation&quot;                           
## [16] &quot;VarScan2&quot;                                     
## [17] &quot;VarScan2 Annotation&quot;                          
## [18] &quot;MuSE&quot;                                         
## [19] &quot;MuSE Annotation&quot;                              
## [20] &quot;BWA with Mark Duplicates and BQSR&quot;            
## [21] &quot;BWA with BQSR&quot;                                
## [22] &quot;VCF LiftOver&quot;                                 
## [23] &quot;MuSE Variant Aggregation and Masking&quot;         
## [24] &quot;MuTect2 Variant Aggregation and Masking&quot;      
## [25] &quot;SomaticSniper Variant Aggregation and Masking&quot;
## [26] &quot;VarScan2 Variant Aggregation and Masking&quot;     
## [27] &quot;FoundationOne Variant Aggregation and Masking&quot;
## [28] &quot;_missing&quot;</code></pre>
</div>
</div>
