---
# Documentation: https://wowchemy.com/docs/managing-content/

title: Database of mRNA gene expression profiles of multiple human organs
subtitle: ''
summary: ''
authors:
- C. G. Son
- S. Bilke
- S. Davis
- B. T. Greer
- J. S. Wei
- C. C. Whiteford
- Q. R. Chen
- N. Cenacchi
- J. Khan
tags: []
categories: []
date: '2005-03-01'
lastmod: 2021-05-08T11:20:31-06:00
featured: false
draft: false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: ''
  focal_point: ''
  preview_only: false

# Projects (optional).
#   Associate this post with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects: []
publishDate: '2021-05-08T17:20:31.318705Z'
publication_types:
- '2'
abstract: Genome-wide expression profiling of normal tissue may facilitate our understanding
  of the etiology of diseased organs and augment the development of new targeted therapeutics.
  Here, we have developed a high-density gene expression database of 18,927 unique
  genes for 158 normal human samples from 19 different organs of 30 different individuals
  using DNA microarrays. We report four main findings. First, despite very diverse
  sample parameters (e.g., age, ethnicity, sex, and postmortem interval), the expression
  profiles belonging to the same organs cluster together, demonstrating internal stability
  of the database. Second, the gene expression profiles reflect major organ-specific
  functions on the molecular level, indicating consistency of our database with known
  biology. Third, we demonstrate that any small (i.e., n approximately 100), randomly
  selected subset of genes can approximately reproduce the hierarchical clustering
  of the full data set, suggesting that the observed differential expression of >90%
  of the probed genes is of biological origin. Fourth, we demonstrate a potential
  application of this database to cancer research by identifying 19 tumor-specific
  genes in neuroblastoma. The selected genes are relatively underexpressed in all
  of the organs examined and belong to therapeutically relevant pathways, making them
  potential novel diagnostic markers and targets for therapy. We expect this database
  will be of utility for developing rationally designed molecularly targeted therapeutics
  in diseases such as cancer, as well as for exploring the functions of genes.
publication: '*Genome Res.*'
---
